frontend/src/metabase/components/EntityMenu.jsx

```diff
@@ -0,0 +1,99 @@
+/* @flow */
+import React, { Component } from 'react'
+import { Motion, spring } from 'react-motion'
+
+import OnClickOutsideWrapper from 'metabase/components/OnClickOutsideWrapper'
+
+import Card from 'metabase/components/Card'
+import EntityMenuTrigger from 'metabase/components/EntityMenuTrigger'
+import EntityMenuItem from 'metabase/components/EntityMenuItem'
+
+type EntityMenuOption = {
+    icon: string,
+    title: string,
+    action?: () => void,
+    link?: string
+}
+
+type Props = {
+    items: Array<EntityMenuOption>,
+    triggerIcon: string
+}
+
+class EntityMenu extends Component {
+
+    props: Props
+
+    state = {
+        open: false
+    }
+
+    toggleMenu = () => {
+        const open = !this.state.open
+        this.setState({ open })
+    }
+
+    render () {
+        const { items, triggerIcon }  = this.props
+        const { open } = this.state
+        return (
+            <div className="relative">
+                <EntityMenuTrigger
+                    icon={triggerIcon}
+                    onClick={this.toggleMenu}
+                    open={open}
+                />
+                { open && (
+                    /* Note: @kdoh 10/12/17
+                     * React Motion has a flow type problem with children see
+                     * https://github.com/chenglou/react-motion/issues/375
+                     * TODO This can be removed if we upgrade to flow 0.53 and react-motion >= 0.5.1
```

Comment:
I added `.*/node_modules/react-motion/.*` to the ignore section `.flowconfig` because I was getting errors related to it when working on one of xrays branches. Not sure if we should do the same here.

Reply:
@attekei Yeah that might be a good place to centralize it. Ideally we can fix this by upgrading our packages at some point.